crates/client/sync/src/commitments/transactions.rs

```diff
@@ -97,13 +94,8 @@ pub async fn memory_transaction_commitment(
     let mut id_builder = BasicIdBuilder::new();
     let id = id_builder.new_id();
 
-    // run in a blocking-safe thread to avoid starving the thread pool
-    let root_hash = spawn_blocking(move || {
-        bonsai_storage.commit(id).expect("Failed to commit to bonsai storage");
-        bonsai_storage.root_hash(identifier).expect("Failed to get root hash")
-    })
-    .await
-    .map_err(|e| format!("Failed to computed transaction root hash: {e}"))?;
+    bonsai_storage.commit(id).expect("Failed to commit to bonsai storage");
```

Comment:
Again, why not run this in the `spawn_blocking` ?

Reply:
Same as above, this function is now called within a rayon context and not tokio